### PR TITLE
Rename package to `django_recaptcha`

### PR DESCRIPTION
**Pull request by Stormheg**
To avoid conflicts with other libraries that use the same package name, like django-simple-captcha.

See: https://github.com/torchbox/django-recaptcha/issues/291

Thanks to @koutoftimer for the name suggestion.

Since we are expecting to make a v4 release soon containing breaking changes, why not include this breaking change too?

This is missing a changelog entry and upgrade considerations. I plan on adding that later when rebasing #312

---

I tested this change by installing my development version of django_recaptcha into a project using it and made the necessary namespace changes. No issues, I think I changed all references.

### CODE REVIEW

**Comment by Stormheg** [comment]
Hey @Andrew-Chen-Wang, could I kindly ask you to review this change?

**Comment by koutoftimer** [comment]
@Stormheg `fields.py:L36` you have missed to change `ImproperlyConfigured` message because it is supposed to show import path new users not familiar with renaming will be confused.

File: django_recaptcha/fields.py
Comment (koutoftimer) [inline django_recaptcha/fields.py:1]: Lines 36-37 should be changes as well

**Review by koutoftimer — CHANGES_REQUESTED** [review]
I have finished proofreading. Little change in error message required.

File: django_recaptcha/apps.py [hunk 1]
```diff
@@ -1,11 +1,11 @@
 from django.apps import AppConfig
 from django.core.checks import Tags, register
 
-from captcha.checks import recaptcha_key_check
+from django_recaptcha.checks import recaptcha_key_check
 
 
-class CaptchaConfig(AppConfig):
-    name = "captcha"
+class DjangoRecaptchaConfig(AppConfig):
```
Comment (Andrew-Chen-Wang) [inline django_recaptcha/apps.py:7]: Code itself doesn't need to change. Just the imports

File: django_recaptcha/apps.py [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
+    name = "django_recaptcha"
```
Comment (Andrew-Chen-Wang) [inline django_recaptcha/apps.py:8]: This should stay the same to avoid package conflict

**Review by Andrew-Chen-Wang — APPROVED** [review]

File: django_recaptcha/fields.py
Comment (Stormheg) [reply django_recaptcha/fields.py:1]: Thanks so much @koutoftimer! Not sure how I missed that instance. Always helpful to have an extra pair of eyes proofreading, much appreciated! 👍

File: django_recaptcha/apps.py (hunk 2 above)
Comment (Stormheg) [reply django_recaptcha/apps.py:8]: Hey @Andrew-Chen-Wang, shouldn't this be the same as the namespace? [The Django docs suggest this should be the full Python path to the package](https://docs.djangoproject.com/en/4.2/ref/applications/#django.apps.AppConfig.name)

If not for technical reason, I still think it should change so the naming is consistent. Won't you agree? 

Also, I'm not sure what package conflict you are referring to. Could you elaborate, please?

File: django_recaptcha/apps.py (hunk 1 above)
Comment (Stormheg) [reply django_recaptcha/apps.py:7]: Figured I changed it to consistently apply the new naming. Would it hurt to keep it like this?

File: django_recaptcha/apps.py (hunk 2 above)
Comment (Andrew-Chen-Wang) [reply django_recaptcha/apps.py:8]: Sorry, I meant django_recaptcha is correct.

**Review by Andrew-Chen-Wang — APPROVED** [review]

**Review by koutoftimer — APPROVED** [review]

**Comment by Stormheg** [comment]
Thanks both! I'll continue with the v4 release once I have some free time on my hands.

**Comment by koutoftimer** [comment]
@Stormheg let me know if you need some help, I'd like this release happening ASAP.

**Pull request merged**